src/com/github/alex_moon/porcupi/views/View.java

```diff
@@ -31,47 +32,45 @@ public View(String name, Controller controller) {
         ManagerServer.get().registerHandler(pokeHandler);
     }
     
-    public String poke(List<String> tokens) {
-        String routeToPoke = tokens.get(0);
+    public String poke(String message) {
+        if (message == null) {
+            return "please specify a route to poke";
+        }
+
+        String routeToPoke = message.split(" ")[0];
         for (String routeFullName: routes.keySet()) {
-            System.out.println("trying to poke " + routeFullName + " with " + routeToPoke);
             if (routeToPoke.equals(routeFullName)) {
+                System.out.println("now poking " + routeFullName + " with " + routeToPoke);
                 poking.add(routeFullName);
-                return "now poking " + routeFullName;
+                return routeFullName;
             }
         }
         return null;
     }
     
-    private void pokeHandle(String routeToPoke, Object track) {
+    private void pokeHandle(String routeToPoke) {
         for (String routeFullName : poking) {
             if (routeToPoke.equals(routeFullName)) {
-                ManagerServer.get().tell("now at " + routeToPoke + " - what would you like to do?");
-                /* synchronized (pokeMonitor) {
-                    try {
-                        // @todo pass pokeMonitor to a "contextHandler" of some kind... 
-                        pokeMonitor.wait();
-                    } catch (InterruptedException e) {
-                        return;
-                    }
-                } */
-                ManagerServer.get().tell(track.toString());
+                pokeHandler.tellOut(
+                    new PokeMessage(routeToPoke, 69696942)
```

Comment:
tbh I'm not even really convinced we need all this bollocks `PokeMessage` and `ContextMessage` and whatever... if it leads to these bogus `tid`s and shit...